### DISCUSSION TITLE
{ name, value } saved as "echo $value > $name"

### DISCUSSION THREAD

**Discussion by hholst80** [nushell/nushell#8516] [Q&A]
Suppose you have a data.json file as:
```json
{
  "name": "test.txt",
  "value": "Y29udGVudHMK"
}
```
and you want to save the decoded data into a file named as the "name" field.

I've tried couple of things but mostly it throws something like "X does not support cell paths" back at me.

```nu
open data.json | do { $in.value | base64 -d | save $in.name }
```

The expected result is that a file called `test.txt` should be created with the content `contents`.

**Comment by fdncred** [comment]
Here's one way.
```
open data.json | do {
  let name = $in.name
  let value = ($in.value | decode base64)
  $value | save $name
}
```
